components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1007,6 +1014,210 @@ public void updateResidentIdP(IdentityProvider identityProvider, String tenantDo
 
     }
 
+    /**
+     * Get all basic identity provider information.
+     *
+     * @param limit        limit per page.
+     * @param offset       offset value.
+     * @param filter       filter value for IdP search.
+     * @param sortOrder    order of IdP ASC/DESC.
+     * @param sortBy       the column value need to sort.
+     * @param tenantDomain tenant domain whose IdP names are requested.
+     * @return Identity Provider's Basic Information array {@link IdpSearchResult}.
+     * @throws IdentityProviderManagementException Error when getting list of Identity Providers.
+     * @throws IOException                         Input error when getting node list.
+     */
+    @Override
+    public IdpSearchResult getIdPs(int limit, int offset, String filter, String sortOrder, String sortBy,
+                                   String tenantDomain) throws IdentityProviderManagementException, IOException {
+
+        if (limit <= 0) {
```

Comment:
shall we move this check to validateLimit() method? WDYT?

Reply:
as we discuss offline, first we check the limit, if the limit is wrong then we do not proceed further. that's why we check the limit at the beginning.